src/content/ui-api/kubectl-gs/login.md

```diff
@@ -109,15 +109,66 @@ kubectl gs login example \
   --certificate-group example-group
 ```
 
-Without `--certificate-group` flag, you can still create a valid certificate, however the client presenting the certificate would not get any permissions in the Kubernetes API. The reason is that without that flag, both the `CN` attribute (interpreted as the user name by the Kubernetes API) and the `O` attribute of the certificate (interpreted as group name) will contain random values, so that no role bindings can match these names.
+#### Certificate lifetime {#certificate-ttl}
 
-For automation, you might want to use a **service account token** to authenticate against the management cluster. Here is an example:
+The certificate created in the above way would have the default lifetime of **one hour**.
+
+You can use the `--certificate-ttl` flag to specify a longer lifetime, e. g. `4h` for four hours. Since client certificates cannot be revoked, we recommend to set the lifetime as short as possible.
+
+#### Groups and users
+
+The generated certificate will have a X.509 subject comprising an `O` field and a `CN` field.
+
+Example:
 
 ```nohighlight
-kubectl gs login example \
+O=73ce775d904da53e,
+CN=73ce775d904da53e.i94nf.k8s.garlic.westeurope.azure.gigantic.io
+```
+
+When presenting such client certificate to the Kubernetes API server, the `CN` field is interpreted as the user identifier, while the `O` field is interpreted as a list of groups the user is a member of.
+
+With these values defaulting to random strings, we ensure that a client presenting this certificate does not accidentally acquire any permissions based on pre-defined group or user names in any (cluster) role bindings.
+
+To assign specific group memberships, or in other terms, to specify additional `O` values in the created certificate, you'll have to use the `--certificate-group` flag when executing the command. This flag can be set multiple times, with one group name each, as shown in the example below.
+
+If `--certificate-group` flag, you can still create a valid certificate, however the client presenting the certificate would not get any permissions in the Kubernetes API. The reason is that without that flag, both the `CN` attribute (interpreted as the user name by the Kubernetes API) and the `O` attribute of the certificate (interpreted as group name) will contain random values, so that no role bindings can match these names.
```

Comment:
I was wondering if the start of the sentence could be a little more specific: "If `--certificate-group` flag" is a bit unclear for me

Reply:
It's definitely bad language :o. Anyway, I basically forgot to adapt that whole paragraph in line 135. Will do.